### ISSUE TITLE
yip should move cursor to start of target if stayOnYank was false

### ISSUE THREAD

**Issue by t9md** [t9md/atom-vim-mode-plus#507]
I didn't notice this mistake since I set `stayOnYank` to true.
When it was `false` cursor should move to start position of target range after operation finished.
This is consistent convention of regular vim operator.
I also have to add spec to verify this behavior.

**Comment by t9md** [comment]
I started investigation by comparing behavior of `g U i p`, `y i p`.
What was strange was
  - `g U i p` **correctly** move cursor to start of paragraph.
  - `y i p` does not move cursor(stay behavior).
  - although both motion have same initial operator properties like `stayOnLinewise`.

The diff was, `g U` mutate text by `selection.insertText()` as a result selection was cleared.
That makes `@target.isLinewise()` become `false`.
But `y` doesn't mutate any text so existing paragraph target return `true` for `@target.isLinewise()` check.

Why `g U i p` correctly move to start of paragraph was by side-effect which was not by intention.

So I re-think, as a result, I conclude the default behavior as following(no `stayOn` option was set).

- basically cursor is moved to start-of-target after operation finished
- but there is some exception such as `y y`, `y 2 y`, `y G`, `g U g U`.
  - what is the commonalities of these operation? it is
    - target is motion and it's wise is **linewise**.
    - (I assumed this part as target is **linewise** in previous version, so linewise text-object is also included, that's why `y i p` behaved stayOn like).

So what I changed here is `stayOnLinewise` operator property now have effect only when target is linewise-motion, not linewise-motion-and-linewise-text-object(yes I excluded text-object from this check.)

So I renamed this important property from old `stayOnLinewise` to `stayOnLinewiseMotion` for explicitness.

**Comment by t9md** [comment]
Things are not so simple as I described above.
I found several exceptional convention.

- When slection is linewise and it's **reversed**(`visual-mode.linewise` and `selection.isReversed()`), it don't move start of target. instead, it land to characterwise selection head(=cursor position), thus stay behavior.
- When not `visual-mode` and motion is linewise and destination was upward direction(=selection is reversed when internally select target), in this case cursor moved to characterwise selection head, (behave same way as in `visual-mode.linewise` and reversed selection situation).

**Comment by t9md** [comment]
> When not visual-mode and motion is linewise and destination was upward direction(=selection is reversed when internally select target), in this case cursor moved to characterwise selection head, (behave same way as in visual-mode.linewise and reversed selection situation).

This rule is still not implemented in vmp.
I have to this behavior consistent in all operator, so will implement by improving mutationTracker.
I want to do some cleanup for mutationTracker, will back soon after that.
